src/stmem.cc

```diff
@@ -376,3 +382,31 @@ mem_hdr::getNodes() const
     return nodes;
 }
 
+void
+mem_hdr::setIdleness(const bool idle)
+{
+    assert(idle != isIdle);
```

Comment:
If this code survives, I suggest relaxing it -- doing nothing when idleness does not change. It may even simplify some callers!